observations/obs_converters/BATS/bats_climatology.py

```diff
@@ -0,0 +1,156 @@
+                   [166, 172]]  # Si1
+
+# layer interface depths for data in prog_z.nc output file from MARBL, run "ncdump -v "z_i" prog_z.nc" to generate.
+marbl_depths = [0, 5, 15, 25, 40, 62.5, 87.5, 112.5, 137.5, 175, 225, 275, 350, 450, 
+                550, 650, 750, 850, 950, 1050, 1150, 1250, 1350, 1450, 1625, 1875, 2250, 
+                2750, 3250, 3750, 4250, 4750, 5250, 5750, 6250]
+
+# set to 'True' in order to produce a histogram of BATS observation vs depth for each month
+plot_sample_hist = False
+
+# human-readable names of the variables being read from the file
+obs_names  = ["O2", "CO2", "Alk", "NO31", "PO41", "Si1"]
+
+#################################################################
+####################### MAIN PROGRAM ############################
+#################################################################
+
+clim_means    = np.zeros((num_obs_types, 12, len(marbl_depths)))
+sq_means      = np.zeros((num_obs_types, 12, len(marbl_depths)))
+sample_counts = np.zeros((num_obs_types, 12, len(marbl_depths)), dtype=int)
+
+bats_data   = open(input_file_path, "r")
+line_number = 0
+
+for line in bats_data.readlines():
+    line_number += 1
+
+    if(line_number < first_data_line):
+        continue
+    
+    # extracting the depth and month where this observation was taken
+
+    depth_val   = float(line[depth_columns[0]-1:depth_columns[1]])
+    month_index = int(line[month_columns[0]-1:month_columns[1]]) - 1
+
+    # assigning this observation to one of the MARBL layers
+
+    depth_index = len(marbl_depths) - 1
+    
+    while((depth_val < marbl_depths[depth_index]) and (depth_index > 0)):
+        depth_index -= 1
+    
+    # adding this observation into the climatology
+
+    for obs_type_index in range(num_obs_types):
+        obs_val = float(line[obs_val_columns[obs_type_index][0]-1:obs_val_columns[obs_type_index][1]])
+        
+        if(abs(obs_val + 999.0) > 1e-8):    # this filters out 'missing' values
+            prev_mean    = clim_means[obs_type_index, month_index, depth_index]
+            prev_sqmean  = sq_means[obs_type_index, month_index, depth_index]
+            prev_samples = sample_counts[obs_type_index, month_index, depth_index]
+
+            new_mean   = (prev_samples*prev_mean + obs_val)/(prev_samples + 1)
+            new_sqmean = (prev_samples*prev_sqmean + obs_val**2)/(prev_samples + 1)
+
+            clim_means[obs_type_index, month_index, depth_index] = new_mean 
+            sq_means[obs_type_index, month_index, depth_index]   = new_sqmean
+            
+            sample_counts[obs_type_index, month_index, depth_index] += 1
+
+            print("month =",month_index,", type =",obs_type_index,", depth =",marbl_depths[depth_index],", value =",obs_val)
+
+bats_data.close()
+
+# setting up a CSV file to write the climatology into; this will be read by the DART data converter.
+
+os.system("rm -f "+output_txt_path)
+output_txt = open(output_txt_path, "w")
```

Comment:
this rm -f  seems dangerous.

Doesn't open(output_txt_path, "w") overwrite the file if it exists?

Reply:
Agreed